developers/contracts/meta-peg-out-endpoint-v2-03.md

```diff
@@ -0,0 +1,242 @@
+
+The first step in the peg-out process is to submit a request. This operation involves the `tx-sender` (requester) initiating a peg-out request for a specified amount of a pegged-in token. Upon initiation, the net amount along with fee costs are escrowed by transferring the token and the gas fee token (`token-abtc`) to the contract. These are held until the request is either finalized or revoked.
+
+To proceed, the request must satisfy several validations:
+
+- The pair (token + chain-id) must be approved in the meta registry.
+- The specified amount must exceed the peg-out operation fee for that particular token.
+- The token peg-out operation must be active in the registry (not paused).
+
+Once these validations are met, the operation is registered in the meta registry via the contract `meta-bridge-registry-v2-03`. The function then generates a unique identifier for the request, known as the `request-id`, ensuring traceability and reference for future operations.
+
+##### Parameters
+
+```lisp
+(amount uint)
+(peg-out-address (buff 128))
+(token-trait <ft-trait>)
+(the-chain-id uint)
+```
+
+#### `claim-peg-out`
+
+Following the peg-out request, the next potential step is for the `tx-sender` (claimer) to claim the request. This step is critical, as it marks the request as ready for finalization. This action is executed by calling the function with the `request-id`, along with the `fulfilled-by` parameter, which designates the address responsible for executing the burn blockchain operation.
+
+The claim must satisfy the following validations:
+
+- The request must exist.
+- The token pair must be approved and operational (not paused for peg-out).
+- The request must not be revoked, finalized, or previously claimed.
+
+Once these conditions are met, the final step is to register the claim in the meta registry. This registration includes the claimer, the fulfilled-by address, and the calculated grace period, during which the claimed request will be eligible for finalization.
+
+##### Parameters
+
+```lisp
+(request-id uint)
+(fulfilled-by (buff 128))
+```
+
+#### `finalize-peg-out-on-index`
+
+Finalizing a peg-out is the final step in committing the peg-out process. This function takes in the burn blockchain transaction (Bitcoin) that corresponds to the Stacks layer operation and executes the peg-out request (`request-id`) along with all related token transfers and their corresponding fees. The finalization can be performed by either a peg-in address or a third-party address.
+
+Key validations include:
+- The request must exist.
+- The token pair must be approved and operational (not paused for peg-out).
+- The transaction cannot be indexed to a time before the contract deployment.
+- The transaction's metaprotocol token (BRC-20) must match the requested details (`tick`, `amount`), with `from` and `to` addresses aligning with the fulfilled-by (registered by the claimer) and the peg-out-address (initiated by the requester) respectively.
```

Comment:
I agree. The whole paragraph was revised for readability. Regarding the concept clarification, we will follow the same criteria as the previous edits since it now seems redundant.